runtime/doc/api.txt

```diff
@@ -704,6 +704,8 @@ nvim_eval_statusline({str}, {opts})                   *nvim_eval_statusline()*
           • start: (number) Byte index (0-based) of first character that uses
             the highlight.
           • group: (string) Name of highlight group.
+          • groups: (array) Names of stacked highlight groups (highest
+            priority last).
```

Comment:
With this, would/can we deprecate `group` (now/ever)?
I also considered omitting `groups` from the return value when there is only 1 group, so no stacking, but always including it is probably worth it for simplicity sake.

Reply:
> With this, would/can we deprecate `group` (now/ever)?

Sure. We don't have a "pattern" for deprecating kwargs fields, but I suggest something like: if the deprecated key is used, call `vim.deprecate(..., '0.13')` using `NLUA_EXEC_STATIC`.

> I also considered omitting `groups` from the return value when there is only 1 group, so no stacking, but always including it is probably worth it for simplicity sake.

Predictability is strongly preferred. Polymorphic returns are painful.

Reply:
> but I suggest something like: if the deprecated key is used

I don't think detecting that is feasible though? At best we could maybe slap a `---@deprecated` on the return class, but that is currently a generic `table<string,any>` for all `Dict` return types, and I'm not even sure the deprecated annotation can be used for class fields.

Reply:
Oh sorry I thought "group" was a kwargs field, not a return value.

For deprecating return values we can just mention it in its docstring, for now.

Reply:
> For deprecating return values we can just mention it in its docstring, for now.

OK, so should we do that here?

Reply:
Sure